### PR TITLE
Lavaland Overhaul: Part 3 - Mining Vendor

### PR DESCRIPTION
**Pull request by Super3222**
Another piece of https://github.com/yogstation13/yogstation/pull/1773. These things are falling down as fast as rat'var components

#### Intent of your Pull Request

# Vendor Update

- The Mining Vendor has been refactored into 6 different categories.
  - Tools: Accesssoreis that help them with mining
  - Weapons: Accessories that can help them with mining, but deal a lot of damage.
  - Meds: Accessories that heal them.
  - Drone: Minebots, and minebot upgrades. Don't get it confused with mining drones.
  - Equipment: Accessories that miners can put on, and suit up with.
  - Miscellaneous: Random crap.

There were also cuts in prices which were gauged on how useful it was to a miner, and how frequently would one actually one want to buy one.

- Mining Hardsuit dropped to 2500 (from 4000).
- Jetpack Upgrade dropped to 1000 (from 2000).
- Resonator dropped to 750 (from 800).
- Upgraded Resonator dropped to 1500 (from 2000).
- Mining Drone dropped to 500 (from 800).
- Diamond Pickaxe dropped to 750 (from 1500).

~~I look forward to pruning the mining vendor even more in the future.~~
I WILL prune the mining vendor in the future. Some of these things - absolutely nobody uses or buys.

# Ore Capsule

Ore capsules are single use items that summon ore boxes.
Miners spawn with 1 ore capsule in their backpack.

I've also tweaked oreboxes to make them extremely pressure resistant.

##### Changelog

:cl:
rscadd: Mining Vendor has been separated into 6 different categories.
tweak: Mining Hardsuit dropped to 2500 (from 4000).
tweak: Jetpack Upgrade dropped to 1000 (from 2000).
tweak: Resonator dropped to 750 (from 800).
tweak: Upgraded Resonator dropped to 1500 (from 2000).
tweak: Mining Drone dropped to 500 (from 800).
tweak: Diamond Pickaxe dropped to 750 (from 1500).
rscadd: Miners now spawn with ore box capsules! Use these to summon real ore boxes.
tweak: Ore boxes are more pressure resistant.
/:cl:

### CODE REVIEW

File: code/modules/mining/machine_vending.dm [hunk 1]
```diff
@@ -9,53 +16,57 @@
...
-		new /datum/data/mining_equipment("Soap",                /obj/item/weapon/soap/nanotrasen, 						          		200),
-		new /datum/data/mining_equipment("Laser Pointer",       /obj/item/device/laser_pointer, 				                   		300),
-		new /datum/data/mining_equipment("Alien Toy",           /obj/item/clothing/mask/facehugger/toy, 		                   		300),
+		new /datum/data/mining_equipment("Advanced Scanner",	/obj/item/device/t_scanner/adv_mining_scanner,                     		800, TOOL),
+		new /datum/data/mining_equipment("Hivelord Stabilizer",	/obj/item/weapon/hivelordstabilizer			 ,                     		400, TOOL),
+		new /datum/data/mining_equipment("Shelter Capsule",		/obj/item/weapon/survivalcapsule			 ,                     		400, TOOL),
+		new /datum/data/mining_equipment("Barometer",			/obj/item/device/barometer/mining,										300, TOOL),
+		new /datum/data/mining_equipment("Jaunter",             /obj/item/device/wormhole_jaunter,										750, TOOL),
+		new /datum/data/mining_equipment("One-Use Ore Capsule",	/obj/item/orecapsule,													100, TOOL),
```
Comment (Super3222) [inline code/modules/mining/machine_vending.dm]: Forgot about this. Maybe I should throw it into this part.

**Comment by ShadowDeath6** [comment]
add mining drones

**Comment by Super3222** [comment]
they wouldn't fit lavaland unless they're slow and can't fly.

**Comment by ShadowDeath6** [comment]
can you call them orebox capsules so people don't think it spawns ores, I just now found out what they do after looking at their code

**Comment by Super3222** [comment]
yeah, i can do that. probably tomorrow though.

**Comment by Gintoki1** [comment]
I'm assuming the golem ships vender won't get any discounts

**Comment by ShadowDeath6** [comment]
It uses different vendor items, so it won't be affected

**Comment by Cruix** [comment]
The golem vendor adds things to the normal list, so this will affect it.

**Comment by Super3222** [comment]
Golems are respawn miners. They start with the same vendor, but have more science-related things inside of it. Enjoy your discounts!

**Pull request merged**